src/content/docs/de/guides/upgrade-to/v3.mdx

```diff
@@ -0,0 +1,678 @@
+
+Astro v3.0 gibt für `Astro.cookies.get(key)` nun `undefined` zurück, wenn kein Cookie existiert.
+
+#### Was soll ich tun?
+
+Diese Änderung wird keinen Code unbrauchbar machen, der auf die Existenz des `Astro.cookie`-Objekts überprüft, bevor `Astro.cookies.get(key)` verwendet wird, ist jedoch jetzt nicht mehr erforderlich.
+
+Du kannst ohne probleme jeglichen Code entfernen, der `has()` verwendet, um zu überprüfen, ob der Wert von `Astro.cookies` `undefined` ist:
+
+```js del={1-3} ins={5-7}
+if (Astro.cookies.has(id)) {
+  const id = Astro.cookies.get(id)!;
+}
+
+const id = Astro.cookies.get(id);
+if (id) {
+}
+```
+
+### Geändert: Programmatisches Ausführen der Astro Kommandozeilenschnittstelle 
+
+In Astro v2.x exportierte und startete der Eintrittspunkt des Pakets `"astro"` die Astro CLI direkt. Es wird nicht empfohlen, Astro auf diese Weise in der Praxis auszuführen.
+
+Astro v3.0 entfernt die CLI aus dem Eintrittspunkt und exportiert einen neuen Satz experimenteller JavaScript-APIs, einschließlich `dev()`, `build()`, `preview()` und `sync()`.
+
+#### Was soll ich tun?
+
+Um die Astro Kommandozeilenschnittstelle [programmatisch auszuführen](/de/reference/cli-reference/#advanced-apis-experimental), verwende die neuen experimentellen JavaScript-APIs:
+
+
+```js
+import { dev, build } from "astro";
+
+// Start the Astro dev server
+const devServer = await dev();
+await devServer.stop();
+
+// Build your Astro project
+await build();
+```
+
+### Geändert: Pfade für den Export von internen Astro-API-Einstiegspunkten
+
+In Astro v2.x konntest du interne Astro-APIs aus `astro/internal/*` und `astro/runtime/server/*` importieren.
+
+Astro v3.0 entfernt die beiden Einstiegspunkte zugunsten des vorhandenen Einstiegspunkts `astro/runtime/*`. Zusätzlich wurde ein neuer Export `astro/compiler-runtime` für compiler-spezifischen Laufzeitcode hinzugefügt.
+
+#### Was soll ich tun?
+
+Diese sind Einstiegspunkte für Astro's interne API und sollten sich nicht auf dein Projekt auswirken. Aber wenn du diese Einstiegspunkte verwendest, aktualisiere sie wie unten gezeigt:
+
+
+```js del={1,4,10} ins={2,5,11}
+import 'astro/internal/index.js';
+import 'astro/runtime/server/index.js';
+
+import 'astro/server/index.js';
+import 'astro/runtime/server/index.js';
+```
+
+```js ins={5} del={4}
+import { transform } from '@astrojs/compiler';
+
+const result = await transform(source, {
+  internalURL: 'astro/runtime/server/index.js',
+  internalURL: 'astro/compiler-runtime',
+  // ...
+});
+```
+
+## Community Ressourcen
+
+Du kennst eine gute Quelle für Astro v3.0? [Bearbeite diese Seite](https://github.com/withastro/docs/edit/main/src/content/docs/en/guides/upgrade-to/v3.mdx) und füge unten einen Link hinzu!
```

Comment:
Finde hier persönlich, dass in dem Kontext (obwohl sie quasi Synonyme sind) `Ressource` besser passt als `Quelle`. Aber entscheide du, was dir lieber ist :)
```suggestion
Du kennst eine gute Ressource für Astro v3.0? [Bearbeite diese Seite](https://github.com/withastro/docs/edit/main/src/content/docs/en/guides/upgrade-to/v3.mdx) und füge unten einen Link hinzu!
```